api/column_family.cc

```diff
@@ -1074,19 +1074,12 @@ void set_column_family(http_context& ctx, routes& r, sharded<db::system_keyspace
     });
 
     cf::force_major_compaction.set(r, [&ctx](std::unique_ptr<http::request> req) -> future<json::json_return_type> {
-        auto params = req_params({
-            std::pair("name", mandatory::yes),
-            std::pair("flush_memtables", mandatory::no),
-            std::pair("consider_only_existing_data", mandatory::no),
-            std::pair("split_output", mandatory::no),
-        });
```

Comment:
I actually like the data-defined structure for the api parameters better than the code-based approach since it allows one to clearly understand the interface in a single glance rather than following custom parsing and processing for each parameter.

What is the motivation for getting rid of req_params?

Reply:
> What is the motivation for getting rid of req_params? 

It's in the PR description -- the req_params is mostly dead code, replacing it with plain `req->get_(query|path)_param()` and the newly introduced `validate_bool_x()` gives the same result with much times less code (the whole set -141:+39)

Reply:
With one exception -- the req_params::process() captures "unknown" parameters in the request and returns back 400(Bad Request). This part is dropped by this PR. However:

1. there are only 5 endpoints that work like that and 1 (single) test that checks it
2. req_params duplicates description from api/api-doc/

So I suggest to re-introddce incoming request validation as generic code that's based on api-doc data